scene/resources/skin_tool.h

```diff
@@ -0,0 +1,101 @@
+/**************************************************************************/
+/*  skin_tool.h                                                           */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef SKIN_TOOL_H
+#define SKIN_TOOL_H
+
+#include "core/io/resource.h"
+#include "core/templates/hash_map.h"
+#include "core/templates/hash_set.h"
+#include "core/templates/rb_set.h"
+
+#include "modules/gltf/gltf_defines.h"
+#include "modules/gltf/structures/gltf_node.h"
+#include "modules/gltf/structures/gltf_skeleton.h"
+#include "modules/gltf/structures/gltf_skin.h"
+#include "scene/main/node.h"
+#include "scene/resources/skin.h"
+
+using SkinNodeIndex = int;
+using SkinSkeletonIndex = int;
+
+class SkinTool : public Resource {
+	GDCLASS(SkinTool, Resource);
```

Comment:
SkinTool consists only of static functions and is not exposed to the user, as it is a shared implementation detail of the GLTF and FBX importers. Because it will never be instantiated, it does not need to extend Resource (or Object) and does not require GDCLASS.

```suggestion
class SkinTool {
```

Reply:
Namespace could also work, but namespaces don't have `private:` so I would prefer to keep it a class with static functions

Reply:
I can put it inside of the ModelDocument3D header but that seems awkward.

Reply:
Not a bad idea... ModelDocument3D seems just as good a place as any: it is a set of utility functions for model document implementations